### ISSUE TITLE
std::monostate thoughts

### ISSUE THREAD

**Issue by jbbjarnason** [stephenberry/glaze#260]
So std::monostate does not implement symmetrical transpose, that is there is different behaviour to_json and from_json.

Amended the current tests which show that the problem is not straight forward:

```cpp
   "variant_write_tests"_test = [] {
      std::variant<double, std::string> d = "not_a_fish";
      std::string s{};

      glz::write_json(d, s);
      expect(s == R"("not_a_fish")");

      d = 5.7;
      s.clear();
      glz::write_json(d, s);
      expect(s == "5.7");

      std::variant<std::monostate, int, std::string> m{};
      glz::write_json(m, s);
      expect(s == R"("std::monostate")") << s;

      expect(glz::read_json(m, s) == glz::error_code::none);  // HIGHLIGHT !!!!!!!!!
   };

   "variant_read_"_test = [] {
      std::variant<int32_t, double> x = 44;
      expect(glz::read_json(x, "33") == glz::error_code::none);
      expect(std::get<int32_t>(x) == 33);

      std::variant<std::monostate, int> y{};
      expect(glz::read_json(y, R"("std::monostate")") == glz::error_code::none);   // HIGHLIGHT !!!!!!!!!!!!
   };
```

The first expection passes because the string `std::monostate` is convertible to string so the resulting variant type is std::string

The second expection fails.

This seems to be implemented here, https://github.com/stephenberry/glaze/blob/main/include/glaze/json/read.hpp#L107 
not sure what is wrong.

**Comment by mwalcott3** [comment]
In my opinion, I think std::monostate and nullptr_t and std::nullptr_t should just always be null. And be some sort of subset of nullable_t like always_null_t. That would make things simpler. I'd be willing to make the necessary changes if @stephenberry is fine with this.

**Comment by stephenberry** [comment]
Yeah, I tend to agree @mwalcott3. I think std::monostate should behave like null, which would allow JSON to set a variant to std::monostate if it was part of the variant. I'll certainly merge if you make this change.

**Comment by mwalcott3** [comment]
> Yeah, I tend to agree @mwalcott3. I think std::monostate should behave like null, which would allow JSON to set a variant to std::monostate if it was part of the variant. I'll certainly merge if you make this change.

Yeah I'll make this change later.

**Comment by mwalcott3** [comment]
Been way busier than I anticipated, stuff has been coming up nonstop. I promise I'll get this done tomorrow.

**Comment by mwalcott3** [comment]
Fixed in #272
